src/sync/impl/sync_client.hpp

```diff
@@ -41,21 +41,26 @@ using ReconnectMode = sync::Client::ReconnectMode;
 struct SyncClient {
     sync::Client client;
 
-    SyncClient(std::unique_ptr<util::Logger> logger,
-               ReconnectMode reconnect_mode = ReconnectMode::normal)
-    : client(make_client(*logger, reconnect_mode)) // Throws
-    , m_logger(std::move(logger))
-    , m_thread([this] {
-        if (g_binding_callback_thread_observer)
-            g_binding_callback_thread_observer->did_create_thread();
-
-        auto will_destroy_thread = util::make_scope_exit([&]() noexcept {
-            if (g_binding_callback_thread_observer)
-                g_binding_callback_thread_observer->will_destroy_thread();
-        });
-
-        client.run(); // Throws
-    }) // Throws
+    SyncClient(std::unique_ptr<util::Logger> logger, ReconnectMode reconnect_mode = ReconnectMode::normal)
+        : client(make_client(*logger, reconnect_mode)) // Throws
```

Comment:
It's a bit unfortunate that you've chosen to re-reformat / re-indent the constructor. It completely obscures your change, while also deviating from the dominant style used in the object store.